src/wasm/wasm-binary.cpp

```diff
@@ -2864,12 +2861,12 @@ void WasmBinaryBuilder::readElementSegments() {
       if (usesExpressions) {
         segment->type = getType();
         if (!segment->type.isFunction()) {
-          throwError("Invalid type for an element segment");
+          throwError("Invalid type for a post-MVP element segment");
         }
       } else {
         auto elemKind = getU32LEB();
         if (elemKind != 0x0) {
-          throwError("Only funcref elem kinds are valid.");
+          throwError("Only funcref elem kinds are valid in the MVP.");
```

Comment:
Maybe "The only allowed elemKind is funcref"? This field also appears in non-MVP segment formats.